pages/nakshatra.vue

```diff
@@ -0,0 +1,379 @@
+      this.nkb1 = 'Moola'
+      this.nkb2 = 'Purva Ashadha'
+      this.nkb3 = 'Uttra Ashadha'
+    } else if (this.rasiBoy === '10') {
+      this.nkb1 = 'Uttra Ashadha'
+      this.nkb2 = 'Shravana'
+      this.nkb3 = 'Dhanishta'
+    } else if (this.rasiBoy === '11') {
+      this.nkb1 = 'Dhanishta'
+      this.nkb2 = 'Shatabhisha'
+      this.nkb3 = 'Purva Bhadrapada'
+    } else if (this.rasiBoy === '12') {
+      this.nkb1 = 'Purva Bhadrapada'
+      this.nkb2 = 'Uttra Bhadrapada'
+      this.nkb3 = 'Revathi'
+    }
+    this.shouldDisable()
+  }
+
+  get nakBoy1 (): string {
+    return `${this.nkb1}`
+  }
+
+  get nakBoy2 (): string {
+    return `${this.nkb2}`
+  }
+
+  get nakBoy3 (): string {
+    return `${this.nkb3}`
+  }
+
+  calcGirl () {
+    if (this.rasiGirl === '1') {
+      this.nkg1 = 'Ashwini'
+      this.nkg2 = 'Bharani'
+      this.nkg3 = 'Kirthika'
+    } else if (this.rasiGirl === '2') {
+      this.nkg1 = 'Kirthika'
+      this.nkg2 = 'Rohini'
+      this.nkg3 = 'Mrigasira'
+    } else if (this.rasiGirl === '3') {
+      this.nkg1 = 'Mrigsira'
+      this.nkg2 = 'Ardra'
+      this.nkg3 = 'Punarvasu'
+    } else if (this.rasiGirl === '4') {
+      this.nkg1 = 'Punarvasu'
+      this.nkg2 = 'Pushyami'
+      this.nkg3 = 'Ashlesha'
+    } else if (this.rasiGirl === '5') {
+      this.nkg1 = 'Magha'
+      this.nkg2 = 'Purva Phalguni'
+      this.nkg3 = 'Uttra Phalguni'
+    } else if (this.rasiGirl === '6') {
+      this.nkg1 = 'Uttra Phalguni'
+      this.nkg2 = 'Hastra'
+      this.nkg3 = 'Chitra'
+    } else if (this.rasiGirl === '7') {
+      this.nkg1 = 'Chitra'
+      this.nkg2 = 'Swathi'
+      this.nkg3 = 'Vishakha'
+    } else if (this.rasiGirl === '8') {
+      this.nkg1 = 'Vishakha'
+      this.nkg2 = 'Anuradha'
+      this.nkg3 = 'Jyestha'
+    } else if (this.rasiGirl === '9') {
+      this.nkg1 = 'Moola'
+      this.nkg2 = 'Purva Ashadha'
+      this.nkg3 = 'Uttra Ashdha'
+    } else if (this.rasiGirl === '10') {
+      this.nkg1 = 'Uttra Ashadha'
+      this.nkg2 = 'Shravana'
+      this.nkg3 = 'Dhanishta'
+    } else if (this.rasiGirl === '11') {
+      this.nkg1 = 'Dhanishta'
+      this.nkg2 = 'Shatabhisha'
+      this.nkg3 = 'Purva Bhadrapada'
+    } else if (this.rasiGirl === '12') {
+      this.nkg1 = 'Purva Bhadrapada'
+      this.nkg2 = 'Uttra Bhadrapada'
+      this.nkg3 = 'Revathi'
+    }
+    this.shouldDisable()
+  }
+
+  shouldDisable () {
+    if (this.rasiBoy !== '0' && this.rasiGirl !== '0') {
+      this.allset = true
+    } else {
+      this.allset = false
+    }
+  }
+
+  get nakGirl1 (): string {
+    return `${this.nkg1}`
+  }
+
+  get nakGirl2 (): string {
+    return `${this.nkg2}`
+  }
+
+  get nakGirl3 (): string {
+    return `${this.nkg3}`
+  }
+
+  calculate () {
+    this.fetchData()
+  }
+
+  nakshatraNum (val1: string, val2: string) {
+    /* return ((parseInt(val1) - 1) * 3) + parseInt(val2) */
+    let val = 1
+    if (parseInt(val1) === 1) {
```

Comment:
`parseInt(val1)` is repeated many times. Call one time assign value to variable and reuse.